src/v2/repositories/implementations/CoinGeckoPriceRepository.ts

```diff
@@ -0,0 +1,21 @@
+import axios from 'axios';
+import { Guard } from 'src/v2//common';
+import { IPriceRepository } from 'src/v2/repositories';
+import { injectable } from 'inversify-props';
+
+@injectable()
+export class CoinGeckoPriceRepository implements IPriceRepository {
```

Comment:
Also, I think `ticker` is better than `token id` (the id defined by coingecko) that is used for the function parameter since we might re-use this class to fetch the price for other ERC20 tokens. So the app doesn't need to know the coingecko id to fetch the price.

Reply:
Thank for the comment @impelcrypto I will implement DIA repository and failover in next iteration. Also I will change to ticker.